static/app/stores/persistedStore.tsx

```diff
@@ -0,0 +1,121 @@
+import React, {
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+
+import useApi from 'sentry/utils/useApi';
+import useOrganization from 'sentry/utils/useOrganization';
+import {OnboardingState} from 'sentry/views/onboarding/targetedOnboarding/types';
+
+import OrganizationStore from './organizationStore';
+import {useLegacyStore} from './useLegacyStore';
+
+type PersistedStore = Readonly<{
+  onboarding: Partial<OnboardingState> | null;
+}>;
+
+const DefaultPersistedStore: PersistedStore = {
+  onboarding: null,
+};
+
+type PersistedStoreContextValue = [
+  PersistedStore | null,
+  React.Dispatch<React.SetStateAction<PersistedStore | null>>
+];
+
```

Comment:
I dont think these need to be null. On L46, you can `const [state, setState] = useState<PersistedStore>({});`. The usePersistedCategory will still return `state[category] ?? null`

```suggestion
type PersistedStoreContextValue = [
  PersistedStore,
  React.Dispatch<React.SetStateAction<PersistedStore>>
];

```